### PR TITLE
[PS-1107] User Verification Service Refactor

### PR DESCRIPTION
**Pull request by justindbaur**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [x] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Refactor the api methods from `api.service` into `userVerification-api.service`.

## Code changes

- Move `api.service` methods to `userVerification-api.service`
- Change all imports to use the proper abstraction/service when needed
- Remove all but one use of `any` in any files I had to change.

## Before you submit

```
- [x] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)
```

### CODE REVIEW

File: apps/web/src/app/settings/two-factor-base.component.ts [hunk 1]
```diff
@@ -4,11 +4,21 @@ import { ApiService } from "@bitwarden/common/abstractions/api.service";
...
+import { BaseResponse } from "@bitwarden/common/models/response/baseResponse";
+
+interface AuthResponseBase {
+  secret: string;
+  verificationType: VerificationType;
+}
+
+export interface AuthResponse<T extends BaseResponse> extends AuthResponseBase {
+  response: T;
+}
```
Comment (MGibson1) [inline apps/web/src/app/settings/two-factor-base.component.ts:21]: We own these classes, it doesn't feel like these interfaces belong here.

File: apps/web/src/app/settings/two-factor-webauthn.component.ts [hunk 2]
```diff
@@ -14,7 +14,15 @@ import {
...
-import { TwoFactorBaseComponent } from "./two-factor-base.component";
+import { AuthResponse, TwoFactorBaseComponent } from "./two-factor-base.component";
+
+interface Key {
+  id: number;
+  name: string;
+  configured: boolean;
+  migrated?: boolean;
+  removePromise: Promise<TwoFactorWebAuthnResponse> | null;
+}
```
Comment (MGibson1) [inline apps/web/src/app/settings/two-factor-webauthn.component.ts:25]: This feels like it belongs here a little more, but just wanted to make sure -- Is this used anywhere else?

**Review by MGibson1 — COMMENTED** [review]
General comment is that this PR tends to localize type definitions alongside where they're used.

This is sometimes an OK pattern, but we need to be sure the types make no sense outside of that context. This should be fairly rare, but is this the case here in general?

File: apps/web/src/app/settings/two-factor-base.component.ts (hunk 1 above)
Comment (justindbaur) [reply apps/web/src/app/settings/two-factor-base.component.ts]: Good call, I moved them to the `types` directory.

File: apps/web/src/app/settings/two-factor-webauthn.component.ts (hunk 2 above)
Comment (justindbaur) [reply apps/web/src/app/settings/two-factor-webauthn.component.ts:26]: As far as I can tell no it's not (other than it's own template).

**Review by MGibson1 — APPROVED** [review]

**Pull request merged**